src/components/buttons/back-button.vue

```diff
@@ -3,19 +3,42 @@
     class="fr-btn fr-btn--secondary fr-btn--icon-left fr-icon-arrow-left-line"
     :class="size && size === 'small' ? 'fr-btn--sm' : ''"
     type="button"
+    @click.prevent="goBack"
   >
     <slot>Précédent</slot>
   </button>
 </template>
 
-<script lang="ts">
-export default {
-  name: "BackButton",
-  props: {
-    size: {
-      type: String,
-      default: null,
-    },
+<script setup lang="ts">
+import { defineProps } from "vue"
+import { useRoute, useRouter } from "vue-router"
+import { EventAction, EventCategory } from "@lib/enums/event.js"
+import tracker from "@/plugins/tracker.js"
+
+const props = defineProps({
+  size: {
+    type: String,
+    default: null,
+  },
+  fallback: {
+    type: String,
+    default: null,
```

Comment:
Ici, j'aurais tendance à mettre une route par défaut valide :
```suggestion
    default: '/simulation/resultats',
```